### PR TITLE
Initial implementation of the tc HFSC class

### PR DESCRIPTION
**Pull request by fbegyn**
An implementation of HFSC class for tc. The service curves are implemented in a way that resembles how they are done by the `tc` cli command.

### CODE REVIEW

**Comment by jellonek** [comment]
Can you squash at least last two commits?
Also PTAL on https://github.com/vishvananda/netlink/blob/master/nl/tc_linux_test.go - additional test for new functionality added in this PR would be nice, but i'm not insisting, that can be also added as a followup PR.

**Comment by fbegyn** [comment]
Apparently I didn't properly cherry pick before the PR

I will see if I can get the testing done in a timely manner, but no promises as to when I'll be able to finish them.

File: qdisc_linux.go [hunk 1]
```diff
@@ -375,6 +382,16 @@ func (h *Handle) QdiscList(link Link) ([]Qdisc, error) {
 					if err := parseTbfData(qdisc, data); err != nil {
 						return nil, err
 					}
+				case "hfsc":
+					log.Println(attr.Value)
+					data, err := nl.ParseRouteAttr(attr.Value)
+					log.Println(data)
```
Comment (aboch) [inline qdisc_linux.go]: I believe this was logging during testing. Please remove

Comment (fbegyn) [reply qdisc_linux.go]: You are correct, should be removed now

**Comment by fbegyn** [comment]
Testing is now implemented within the same limits as the existing Class testing. If further tests are required or there are other remarks, please let me know.

**Comment by fbegyn** [comment]
ping @aboch @jellonek

**Comment by aboch** [comment]
Thanks @fbegyn .
Please squash the commits into one.

**Comment by fbegyn** [comment]
@aboch squashed. Need to get into the habit of doing that before issuing PRs.

**Comment by aboch** [comment]
LGTM

**Pull request merged**